runtime/lua/vim/lsp/client.lua

```diff
@@ -395,11 +401,11 @@ function Client.create(config)
     capabilities = config.capabilities,
     workspace_folders = lsp._get_workspace_folders(config.workspace_folders or config.root_dir),
     root_dir = config.root_dir,
```

Comment:
If `root_dir=false`, where does that actually get resolved to "default impl"? Or same question for `root_markers=false.` I assume that's not a default behavior of `vim.tbl_deep_extend()` ...

Reply:
If `root_dir=false`, then `root_markers` are used. If they are also `false|nil`, then no root dir is determined. It's the same as if the config omitted these fields.